### PR TITLE
Fix impl and tests of connection after megring singletones

### PR DESCRIPTION
**Pull request by Kozoriz**
Corrected Connection class constructor
Corrected connection_test, connection_handler_test.

### CODE REVIEW

**Comment by Kozoriz** [comment]
please review : @AByzhynar , @LuxoftAKutsan , @VVeremjova

File: src/components/connection_handler/test/connection_handler_impl_test.cc [hunk 1]
```diff
@@ -102,6 +102,8 @@ class ConnectionHandlerTest : public ::testing::Test {
                                                     device_name_,
                                                     connection_type_);
     // Add Device and connection
+    ON_CALL(mock_connection_handler_settings, heart_beat_timeout())
+        .WillByDefault(Return(1000));
```
Comment (VVeremjova) [inline src/components/connection_handler/test/connection_handler_impl_test.cc]: It should be 1000u

File: src/components/connection_handler/test/connection_test.cc [hunk 2]
```diff
@@ -59,10 +59,11 @@ class ConnectionTest : public ::testing::Test {
         mock_connection_handler_settings, transport_manager_mock);
     const ConnectionHandle connectionHandle = 0;
     const DeviceHandle device_handle = 0;
+    const uint32_t heart_beat = 10000;
```
Comment (VVeremjova) [inline src/components/connection_handler/test/connection_test.cc]: 10000u

**Comment by AByzhynar** [comment]
@Kozoriz
1) Please change `<gmock.h>` -> `"gmock.h"`
2) In Connection Ctor initializer list please make all in one style.

File: src/components/connection_handler/test/connection_test.cc [hunk 3]
```diff
@@ -175,6 +176,36 @@ TEST_F(ConnectionTest, HeartBeat_Supported) {
...
+
+  const ConnectionHandle connectionHandle = 0;
+  const DeviceHandle device_handle = 0;
+  const uint32_t heart_beat = 0;
+  connection_ = new Connection(connectionHandle,
+                               device_handle,
+                               connection_handler_,
+                               heart_beat);
+  StartSession();
+  // Check if protocol version is 4
```
Comment (VVeremjova) [inline src/components/connection_handler/test/connection_test.cc]: Where is  the check? I see only version update

Comment (Kozoriz) [inline src/components/connection_handler/test/connection_test.cc]: @VVeremjova , this comment about checking "support" after setting protocol to V4

**Comment by Kozoriz** [comment]
@AByzhynar , @VVeremjova see  		b778480

**Comment by AByzhynar** [comment]
Reviewed

**Comment by VVeremjova** [comment]
Reviewed

**Comment by malirod** [comment]
Reviewed

**Pull request merged**